### PR TITLE
allow reference strip to work in a web component

### PR DESCRIPTION
**Pull request by dmcalpin**

### CODE REVIEW

**Comment by dmcalpin** [comment]
fixes #1674

**Comment by dmcalpin** [comment]
the unit test fails because of phantomjs. updating `grunt-contrib-qunit` to version 3+ would fix this since it uses headless chrome

File: src/referencestrip.js [hunk 1]
```diff
@@ -236,7 +236,8 @@ $.extend( $.ReferenceStrip.prototype, $.EventSource.prototype, $.Viewer.prototyp
      * @function
      */
     setFocus: function ( page ) {
-        var element      = $.getElement( this.element.id + '-' + page ),
+        var element      = this.element.querySelector('#' + this.element.id + '-' + page ),
+        // var element      = $.getElement( this.element.id + '-' + page ),
```
Comment (iangilman) [inline src/referencestrip.js]: Please remove the commented out line.

File: test/modules/referencestrip.js [hunk 2]
```diff
@@ -44,4 +44,33 @@
         done();
     });
 
+    // ----------
+    QUnit.test('shadow dom', function(assert) {
+        var done = assert.async();
+
+        class ShadowElem extends HTMLElement {
```
Comment (iangilman) [inline test/modules/referencestrip.js]: Interesting! Is it possible to test the shadow DOM without using ES6? We technically haven't sunset IE8 support (though we're not that diligent about it), so we haven't been introducing modern browser features (except Canvas).

**Comment by dmcalpin** [comment]
i'll get started on the qunit PR, then I'll clean this up

**Comment by iangilman** [comment]
Great, thank you!

File: test/modules/referencestrip.js (hunk 2 above)
Comment (dmcalpin) [reply test/modules/referencestrip.js]: I was able to remove the use of `class`, and just attach a shadowRoot to a `div`, this still means that the test will fail in IE11 and below. No other way to test the viewer inside shadowDOM.

do you value the test proving it works in shadowDOM, or having the suite working in IE11 😄 Let me know. i'll make adjustments if you prefer i remove the test. If that is the route though, I think adding a comment in `src/referencestrip.js` explaining why we use `this.element...` should be added so the code isnt accidentally updated in the future

**Comment by iangilman** [comment]
Yeah, I'm not that concerned about making sure the tests run in IE11, though it would be nice. Would it be possible to do a userAgent sniff and disable that one test on IE?

**Comment by dmcalpin** [comment]
Off the top of my head I’m not sure, but will look into it

**Comment by iangilman** [comment]
Looks like there's detection built into OSD... You could do something like:

```
if (OpenSeadragon.Browser.vendor === OpenSeadragon.BROWSERS.IE && version <= 11) {
  return;
}
```

**Comment by dmcalpin** [comment]
how do you feel about feature detection instead, this way other legacy browsers are covered as well
`if (document.head && document.head.attachShadow) {` will detect if attachShadow is available, and `document.head` is supported as far back as IE9

**Comment by msalsbery** [comment]
@iangilman

What minimum browser versions are we now supporting? May affect what I can now do with pointer event work ;) Probably not but worth asking!

**Comment by iangilman** [comment]
@dmcalpin Yes, feature detection is better! Looks beautiful... Thank you for making this happen!

@msalsbery Good question! Looks like we have officially let IE8 go (now that I review https://github.com/openseadragon/openseadragon/issues/373). I think we should continue to support IE11, but I bet we don't need IE9 or 10. I guess that's another question to raise with the community. Also, I think we still haven't made an official statement anywhere about what browsers we support...

**Pull request merged**